public/js/application.js

```diff
@@ -1 +1,144 @@
\ No newline at end of file
+var $input = $('<div class="modal-body"><input type="text" class="form-control" placeholder="Message"></div>');
+$(document).on("click", ".js-msgGroup", function () {
+    $(".js-msgGroup, .js-newMsg").addClass("hide"), $(".js-conversation").removeClass("hide"), $(".modal-title").html('<a href="#" class="js-gotoMsgs">Back</a>'), $input.insertBefore(".js-modalBody")
+}), $(function () {
+    function o() {
+        return $(window).width() - ($('[data-toggle="popover"]').offset().left + $('[data-toggle="popover"]').outerWidth())
+    }
+
+    $(window).on("resize", function () {
+        var t = $('[data-toggle="popover"]').data("bs.popover");
+        t && (t.options.viewport.padding = o())
+    }), $('[data-toggle="popover"]').popover({
+        template: '<div class="popover" role="tooltip"><div class="arrow"></div><div class="popover-content p-x-0"></div></div>',
+        title: "",
+        html: !0,
+        trigger: "manual",
+        placement: "bottom",
+        viewport: {selector: "body", padding: o()},
+        content: function () {
+            var o = $(".app-navbar .navbar-nav:last-child").clone();
+            return '<div class="nav nav-stacked" style="width: 200px">' + o.html() + "</div>"
+        }
+    }), $('[data-toggle="popover"]').on("click", function (o) {
+        o.stopPropagation(), $('[data-toggle="popover"]').data("bs.popover").tip().hasClass("in") ? ($('[data-toggle="popover"]').popover("hide"), $(document).off("click.app.popover")) : ($('[data-toggle="popover"]').popover("show"), setTimeout(function () {
+            $(document).one("click.app.popover", function () {
+                $('[data-toggle="popover"]').popover("hide")
+            })
+        }, 1))
+    })
+}), $(document).on("click", ".js-gotoMsgs", function () {
```

Comment:
Line is too long.<br>Mixed double and single quotes.<br>Too many errors. (21% scanned).